src/commands/docs.ts

```diff
@@ -144,151 +144,148 @@ const command: Command = {
 		return true;
 	},
 	async execute(client) {
-		// Deferred async code can not be moved outside of a Promise
-		client.ctx.waitUntil(
-			new Promise(async (resolve) => {
-				let query = getStringOption(client.interaction.data, 'query')!;
-
-				if (query.startsWith('auto-')) {
-					const reply: SearchHit = await index.getObject(query.substring(5));
-					await returnObjectResult(client.interaction, reply, client.env);
-					return;
-				}
+		client.ctx.waitUntil((async () => {
```

Comment:
I think @alexanderniebuhr's suggestion to remove the IIFE is a good one!

Maybe we could refactor this to a `client.waitUntil` function that accepts an async function? It would probably call the function and pass it directly to `ctx.waitUntil`.